src/environment/Permittivity.cpp

```diff
@@ -30,81 +30,22 @@
 namespace mrchem {
 
 Permittivity::Permittivity(const mrchem::Cavity cavity, double epsilon_in, double epsilon_out, std::string formulation)
-        : epsilon_in(epsilon_in)
-        , epsilon_out(epsilon_out)
-        , formulation(formulation)
-        , cavity(cavity) {}
+        : StepFunction(cavity, epsilon_in, epsilon_out) {}
 
 double Permittivity::evalf(const mrcpp::Coord<3> &r) const {
```

Comment:
Maybe instead of changing the state of the class (`this->inverse`) before calling `evalf`, you can add a boolean to the function?
```suggestion
double Permittivity::evalf(const mrcpp::Coord<3> &r, bool inverse) const {
```

Reply:
Maybe it violates inheritance though 😞

Reply:
This seems like a nightmare to implement. I think I need to make overloads for the apply and multiply methods of mrcpp for that to work

Reply:
Turns out this was actually an easy fix with lambda functions :/

Reply:
Hail the lambda functions! 🎊 $\lambda$!